### PR TITLE
feat: add parameter to clone cells-site-template

### PR DESCRIPTION
**Pull request by enicio**

### CODE REVIEW

**Review by jotanarciso — DISMISSED** [review]

File: src/build.ts [hunk 1]
```diff
@@ -36,24 +37,18 @@ export class Builder {
         this.targetDir = targetDir;
     }
 
-    static init(): Builder {
-        const targetDir = process.cwd();
-        try {
-            const workerDir = path.join(targetDir, "worker");
+    static createWorkerDir(workerTargetDir: string) {
+        const workerDir = path.join(workerTargetDir, "worker");
```
Comment (lmalheiro) [inline src/build.ts:41]: I find suspicious that we have a constructor with parameter `targetDir` and then a static method with parameter `workerTargetDir`. Shouldn't it be an instance method and reuse the constructor parameter?

File: src/build.ts [hunk 2]
```diff
@@ -173,14 +168,46 @@ export class Builder {
...
                 await builder.buildClient();
+
+                Builder.createWorkerDir(targetDir);
+                manifest = builder.generateManifest(options.assetsDir);
+
                 webpackConfigPath = WORKER_CFG_PATH;
+            } else {
+                // checking static site template
+                const templatePath = 'azion/cells-site-template'
+                const staticSiteWorkerDir = path.join(targetDir, templatePath);
```
Comment (lmalheiro) [inline src/build.ts:187]: Maybe `templatePath` should be a constructor parameter, then the constructor could compute the worker dir as a private instance property and `createWorkerDir()` would be an instance method. `isInitTemplate` could become an instance method too.
That could simply this `if else` statement.

File: src/build.ts [hunk 3]
```diff
@@ -173,14 +168,46 @@ export class Builder {
...
             if (!options.staticSite) {
                 await builder.buildClient();
+
+                Builder.createWorkerDir(targetDir);
+                manifest = builder.generateManifest(options.assetsDir);
+
                 webpackConfigPath = WORKER_CFG_PATH;
+            } else {
+                // checking static site template
+                const templatePath = 'azion/cells-site-template'
```
Comment (lmalheiro) [inline src/build.ts:186]: Is `'azion/cells-site-template'` a constant that can be used at the `asset-publisher.ts` too?

File: src/asset-publisher.ts [hunk 4]
```diff
@@ -38,15 +39,23 @@ export class AssetPublisher {
...
         this.s3 = s3;
+        this.staticSite = staticSite
     }
 
     public async deployStaticAssets(subdir = 'out') {
         const waitList = [];
-        const manifest = new ManifestBuilder(this.rootPath).loadManifest();
+
+        let manifest;
+        if(this.staticSite) {
```
Comment (lmalheiro) [inline src/asset-publisher.ts:53]: `staticSite` being a boolean indicates that we could have a base `AssetPublisher` and specialize it overloading the `deployStaticAssets()` method.

File: src/init.ts [hunk 5]
```diff
@@ -15,6 +18,8 @@ import {
     NotADirectory
 } from './errors';
 
+const cellSiteTemplateRepo = "https://github.com/aziontech/cells-site-template.git";
```
Comment (lmalheiro) [inline src/init.ts:21]: Shouldn't the constants be uppercase?

File: src/init.ts [hunk 6]
```diff
@@ -75,8 +80,38 @@ async function init(targetDir: string, repository: string, options: any): Promis
 
 }
 
+async function initCellsTemplate(targetDir: string, cellSiteTemplateRepo: string) {
+    try {
+        console.log("Creating azion directory");
+        if(!fs.existsSync("azion")) fs.mkdirSync("azion");
+
+        const repoDir = path.join(targetDir, "azion/cells-site-template");
```
Comment (lmalheiro) [inline src/init.ts:88]: Shouldn't `"azion/cells-site-template"` be a constant?

File: src/init.ts [hunk 7]
```diff
@@ -75,8 +80,38 @@ async function init(targetDir: string, repository: string, options: any): Promis
 
 }
 
+async function initCellsTemplate(targetDir: string, cellSiteTemplateRepo: string) {
```
Comment (lmalheiro) [inline src/init.ts:83]: Parameter `cellSiteTemplateRepo` overrides const `cellSiteTemplateRepo` declared at the beginning of the file.

Comment (augustollenz) [reply src/init.ts:83]: This comment is now obsolete.

**Review by lmalheiro — APPROVED** [review]

**Review by MagnunAVFAzion — APPROVED** [review]

**Pull request merged**